validation/validate.ts

```diff
@@ -0,0 +1,329 @@
+import {CDPSession} from 'puppeteer';
+import * as puppeteer from 'puppeteer';
+import {Protocol} from 'devtools-protocol';
+
+import {bgPrototypeSource} from './accname_libs_source_code/bg_prototype_source';
+import {ourLibSource} from './accname_libs_source_code/our_lib_source';
+import * as axe from 'axe-core';
+
+const inputHTML = `
+  <html>
+    <head>
+      <title>
+        A Sample HTML Document
+      </title>
+    </head>
+    <body>
+      <div id="bar">world</div>
+      <div id="baz">Hello</div>
+      <div id="foo" aria-labelledby="baz bar"></div>
+    </body>
+  </html>
+`;
+
+(async () => {
+  const browser = await puppeteer.launch({
+    args: ['--enable-blink-features=AccessibilityObjectModel'],
+  });
+  const page = await browser.newPage();
+  await page.goto('data:text/html,' + inputHTML);
+
+  const client = await page.target().createCDPSession();
+
+  await loadAccNameLibraries(client);
+
+  const chromeAccName = await getChromeAccName(client, 'foo');
+  console.log('Chrome AccName: ', chromeAccName);
+
+  const axeAccName = await getAXEAccName(client, 'foo');
+  console.log('aXe AccName: ', axeAccName);
+
+  const aomAccName = await getAOMAccName(client, 'foo');
+  console.log('AOM AccName: ', aomAccName);
+
+  const bgAccName = await getBGAccName(client, 'foo');
+  console.log('BG AccName: ', bgAccName);
+
+  const ourAccName = await getOurAccName(client, 'foo');
+  console.log('Our AccName: ', ourAccName);
+
+  const minimisedHTML = await getMinimisedHTML(client, 'foo');
+  console.log('\nHTML Snippet used by Chrome:\n', minimisedHTML);
+
+  await browser.close();
+})();
+
+/*
+
+  Functions for loading and running the various AccName implementations
+
+*/
+
+async function loadAccNameLibraries(client: CDPSession) {
+  // Load axe-core
+  await client.send('Runtime.evaluate', {expression: axe.source});
```

Comment:
This would be ideal but unfortunately (as far as I can tell) we can't get HTMLElements from puppeteers api, instead it returns [ElementHandles](https://pptr.dev/#?product=Puppeteer&version=v5.2.1&show=api-class-elementhandle). Also the implementations use the DOM api (`document`, `window` etc) so it seems easiest just to run them in the browser.

I'm also scared of passing source code by string but I think this is _almost_ how the libraries are intended to be used (https://github.com/dequelabs/axe-core#getting-started) (https://github.com/WhatSock/w3c-alternative-text-computation#how-to-import). If passing the js into the console is equivalent to embedding it with a `<script>` tag then maybe that's okay? If not we could actually append script tags to the page using puppeteer easily enough. What do you think?

Reply:
See the comment below - if you can treat an `ElementHandle` like an `Element` through `page.evaluate` does that help?